test/minikube/minikube_base_insights_test.go

```diff
@@ -0,0 +1,199 @@
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+
+	options := helm.Options{
+		SetValues: map[string]string{
+			"insights.enabled":                      "true",
+			"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			// Load custom user plugin for the admin
+			"insights.plugins.admin.log": `
+[[inputs.tail]]
+  files = ["/var/log/nuodb/nuoadmin*.log"]
+  from_beginning = true
+  name_override= "logfile"
+  data_format = "grok"
+  grok_patterns= [ "%{CUSTOM_LOGLINE}" ]
+  grok_custom_patterns = '''
+  CUSTOM_LOGLINE %{TIMESTAMP_ISO8601:timestamp:ts-"2006-01-02T15:04:05.000-0700"}%{SPACE}(?:%{LOGLEVEL:loglevel:tag}%{SPACE}(?:%{NOTSPACE:logger:tag}%{SPACE})?)?%{GREEDYDATA:message}
+  '''
+  [inputs.tail.tags]
+    db_tag = "nuolog"`,
+		},
+	}
+
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+	adminReleaseName, namespaceName := testlib.StartAdmin(t, &options, 1, namespaceName)
+	admin0 := fmt.Sprintf("%s-nuodb-cluster0-0", adminReleaseName)
+
+	t.Run("startDatabaseStatefulSet", func(t *testing.T) {
+		defer testlib.Teardown(testlib.TEARDOWN_DATABASE) // ensure resources allocated in called functions are released when this function exits
+
+		databaseRealeaseName := testlib.StartDatabase(t, namespaceName, admin0, &options)
+		createOutputFilePlugin(t, namespaceName)
+		defer testlib.Teardown(testlib.TEARDOWN_INSIGHTS)
+		testlib.PopulateDBWithQuickstart(t, namespaceName, admin0, "demo")
+		t.Run("verifyMetricsCollection", func(t *testing.T) {
+			verifyCollectionForAdmin(t, namespaceName, fmt.Sprintf("%s-nuodb-cluster0", adminReleaseName))
+			verifyCollectionForDatabase(t, namespaceName, fmt.Sprintf("%s-nuodb-%s-%s", databaseRealeaseName, "cluster0", "demo"), "demo")
+		})
+	})
+
+	t.Run("startDatabaseDaemonSet", func(t *testing.T) {
+		defer testlib.Teardown(testlib.TEARDOWN_DATABASE) // ensure resources allocated in called functions are released when this function exits
+		options.SetValues["database.enableDaemonSet"] = "true"
+		// Start only hotcopy SM daemonset
+		options.SetValues["database.sm.noHotCopy.enablePod"] = "false"
+		databaseRealeaseName := testlib.StartDatabase(t, namespaceName, admin0, &options)
+
+		createOutputFilePlugin(t, namespaceName)
+		defer testlib.Teardown(testlib.TEARDOWN_INSIGHTS)
+		testlib.PopulateDBWithQuickstart(t, namespaceName, admin0, "demo")
+		t.Run("verifyMetricsCollection", func(t *testing.T) {
+			verifyCollectionForAdmin(t, namespaceName, fmt.Sprintf("%s-nuodb-cluster0", adminReleaseName))
+			verifyCollectionForDatabase(t, namespaceName, fmt.Sprintf("%s-nuodb-%s-%s", databaseRealeaseName, "cluster0", "demo"), "demo")
+		})
+	})
+
+	t.Run("startDatabaseMultiTenant", func(t *testing.T) {
```

Comment:
this sub-test probably takes forever and a half. If this test would be pushing data from multiple sources to the same visualization product, I would say "lets keep it" in order to verify that the visualization product can behave in a correct multi-tenant fashion. But the collector being tested here should not do anything special if there are multiple independent DBs running. So I suggest that we don't have this test.

Reply:
Locally it takes ~5 min:
```
--- PASS: TestInsightsMetricsCollection (337.55s)
    --- PASS: TestInsightsMetricsCollection/startDatabaseMultiTenant (271.38s)
        --- PASS: TestInsightsMetricsCollection/startDatabaseMultiTenant/verifyMetricsCollectionForAdmin (24.01s)
        --- PASS: TestInsightsMetricsCollection/startDatabaseMultiTenant/verifyMetricsCollectionForGreen (25.80s)
        --- PASS: TestInsightsMetricsCollection/startDatabaseMultiTenant/verifyMetricsCollectionForBlue (18.02s)
```

The main reason to have this test is to verify that multiple DB deployments can forward metrics to a single destination. I'm fine skipping the test to save test execution time.

Reply:
ok with me. resolving